secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/payment/services/validation/RiskValidationService.java

```diff
@@ -23,6 +23,8 @@
 import uk.org.openbanking.datamodel.common.OBExternalPaymentContext1Code;
 import uk.org.openbanking.datamodel.common.OBRisk1;
 
+import java.util.Objects;
+
 @Service
 @Slf4j
 public class RiskValidationService {
```

Comment:
This class does not appear to be validating all of the risk fields of the consent vs the payment.

I think it should also use .equals to do a full check

Reply:
The risk validation that's is implemented currently must be changed to use .equals for the next release, to use the same logic that we have to validate the initiation.